src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -911,10 +915,70 @@ protected override void AddExistingDocument(DocumentId documentId, string filePa
                 throw new Exception(ServicesVSResources.Could_not_find_location_of_folder_on_disk);
             }
 
-            return AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+            AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+        }
+
+        private bool TryAddEditorConfigToSolutionItems(
+            DocumentInfo documentInfo,
+            SourceText text)
+        {
+            // We're going to see if this is an .editorconfig being added at the solution level, and if so process it specially; so verify first that it is one
+            if (CurrentSolution.FilePath is null || documentInfo.FilePath is null)
+                return false;
+
+            if (PathUtilities.GetFileName(documentInfo.FilePath) != ".editorconfig")
+                return false;
+
+            if (IOUtilities.PerformIO(() => File.Exists(documentInfo.FilePath)))
```

Comment:
doc?

Reply:
I can't explain why this is checked -- this is preserving behavior of the previous system. @mavasani any idea why there might be a check?

Reply:
I believe it was defensive check - in case creating a new editorconfig by the code fix application failed, we want to avoid attempting to create a Solution Items folder.